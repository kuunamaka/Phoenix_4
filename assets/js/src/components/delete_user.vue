<template>
  <button class="delete" @click="deleteUser(user)">Delete</button>
</template>

<script>
import axios from 'axios'
export default {
  name: 'delete_user',
  data() {
    return{
      user: ''
    }
  },
  methods: {
    async deleteUser(user) {
      try {
        const resp = await axios.delete(`/api/users` + `/${ user.id }`, {
          user: {
            name: user.name,
            username: user.username
          }
        })
        console.log(resp);
        window.location.href = '/users';
      }
      catch (e) {
        console.log(e)
      }
    }
  },
  async mounted() {
    try {
      const user_id = window.location.pathname.split('/')[2];
      const resp = await axios.get('/api' + `/users/${ user_id }`)
      this.user = resp.data
    }
    catch (e) {
      console.log(e)
    }
  }
}
</script>
