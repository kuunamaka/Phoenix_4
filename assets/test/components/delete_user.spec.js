import { shallowMount } from '@vue/test-utils'
import delete_user from  '@/src/components/delete_user.vue'

const axios = require("axios");
const MockAdapter = require("axios-mock-adapter");
const mock = new  MockAdapter(axios);

mock.onGet("/api/users/1").reply(200, {
    id: 1, name: "Test", username: "testtest"}
);

global.window = Object.create(window);
const pathname = '/users/1';

Object.defineProperty(window, 'location', {
  value: {
    pathname,
    href: `http://localhost:4001${ pathname }`
  }
});

describe('delete_user.vue', () => {
  const wrapper = shallowMount(delete_user);
  const vm = wrapper.vm
    
  it('is a Vue instance', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  
  describe('deleteUser()', () => {
    it('redirect to /users', async () => { 
      mock.onDelete('/api/users/1').reply(200, {user: {id: 1, name: 'Test', username: 'testtest'}});
      await wrapper.setData({user: {id: 1, name: 'Test', username: 'testtest'}})
  
      await vm.deleteUser({id: 1, name: 'Test', username: 'testtest'})
  
      expect(window.location.href).toEqual('/users');
    });
  })
})
