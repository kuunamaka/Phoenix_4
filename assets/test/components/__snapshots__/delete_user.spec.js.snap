// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete_user.vue is a Vue instance 1`] = `
<button
  class="delete"
>
  Delete
</button>
`;
